### PR TITLE
[flutter_tools] Include more details in structured errors sent to a DAP client

### LINKED ISSUE #4743: Bad structured errors text
Using the code at https://github.com/DanTup/devtools/tree/bad-structured-errors, running the sidebar results in this error that appears to be missing information:

image

Specifically, it says "The relevant error-causing widget was:" but has nothing after it.

### PR DESCRIPTION
**Pull request by DanTup**
The debug adapter converts Flutter's structured errors into a text format to be sent to the debug client and shown in the console. When an error is not the first error since the last reload, it is shown as just a summary (since it may be caused by a prior error). In this mode, the filter was causing some important information (the erroring widget) to be omitted.

This tweaks the logic to include child nodes of a `DiagnosticBlock` in this mode.

Fixes https://github.com/Dart-Code/Dart-Code/issues/4743

## Before:

image

## After:

image

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter_tools/lib/src/debug_adapters/error_formatter.dart [hunk 1]
```diff
@@ -118,7 +118,11 @@ class FlutterErrorFormatter {
     final bool allChildrenAreLeaf = node.children.isNotEmpty &&
         !node.children.any((_ErrorNode child) => child.children.isNotEmpty);
     if (node.level == _DiagnosticsNodeLevel.summary || allChildrenAreLeaf) {
-      _writeNode(node, recursive: false);
+      // DiagnosticsBlock is a container, so recursive into its children if
```
Comment (andrewkolos) [inline packages/flutter_tools/lib/src/debug_adapters/error_formatter.dart:121]: nit: typo?
```suggestion
      // DiagnosticsBlock is a container, so recurse into its children if
```

**Review by andrewkolos — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/debug_adapters/error_formatter.dart (hunk 1 above)
Comment (DanTup) [reply packages/flutter_tools/lib/src/debug_adapters/error_formatter.dart]: Thanks, fixed!

**Comment by DanTup** [comment]
Thanks!

**Pull request merged**